presto-hive/src/test/java/com/facebook/presto/hive/statistics/TestParquetQuickStatsBuilder.java

```diff
@@ -182,29 +186,24 @@ private ImmutableList<HiveFileInfo> buildHiveFileInfos(String basePath, String p
     @BeforeTest
     private void setUp()
     {
-        metastoreMock = Mockito.mock(SemiTransactionalHiveMetastore.class);
-        Storage mockStorage = new Storage(
-                fromHiveStorageFormat(PARQUET),
-                "some/path",
-                Optional.empty(),
-                true,
-                ImmutableMap.of(),
-                ImmutableMap.of());
-        Partition mockPartition = new Partition(
+        Table table = new Table(
                 TEST_SCHEMA,
                 TEST_TABLE,
+                "owner",
+                MANAGED_TABLE,
+                Storage.builder()
+                        .setStorageFormat(fromHiveStorageFormat(PARQUET))
+                        .setLocation("location")
+                        .build(),
                 ImmutableList.of(),
-                mockStorage,
                 ImmutableList.of(),
                 ImmutableMap.of(),
                 Optional.empty(),
-                false,
-                true,
-                0,
-                0,
                 Optional.empty());
-        when(metastoreMock.getPartition(any(), eq(TEST_SCHEMA), eq(TEST_TABLE), any()))
-                .thenReturn(Optional.of(mockPartition));
+
+        TestingSemiTransactionalHiveMetastore mock = TestingSemiTransactionalHiveMetastore.create();
+        mock.addTable(TEST_SCHEMA, TEST_TABLE, table, ImmutableList.of());
+        metastore = mock;
```

Comment:
only need one variable here

Reply:
+1, maybe make the member variable a `TestingSemiTransactionalHiveMetastore`

Reply:
The choice here was intentional, I wanted to have tests limit access to the interface `SemiTransactionalHiveMetastore`